test/index.test.js

```diff
@@ -0,0 +1,241 @@
+    server.route({
+      method: 'GET',
+      path: '/genres',
+      config: {
+        plugins: {
+          totalCount: { model: Genre }
+        },
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/genres?include[]=total_count',
+      credentials: {}
+    })
+    .then((res) => {
+      expect(res.result.total_count).to.eql(3);
+    });
+  });
+
+  it('appends the total when used with the hapi-query-filter plugin', () => {
+    server.route({
+      method: 'GET',
+      path: '/books',
+      config: {
+        plugins: {
+          queryFilter: { enabled: true },
+          totalCount: { model: Book }
+        },
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/books?year=1984&include[]=total_count',
+      credentials: {}
+    })
+    .then((res) => {
+      expect(res.result.total_count).to.eql(2);
+    });
+  });
+
+  it('appends the unfiltered total when used without the hapi-query-filter plugin', () => {
+    server.route({
+      method: 'GET',
+      path: '/genres',
+      config: {
+        plugins: {
+          queryFilter: { enabled: false },
+          totalCount: { model: Genre }
+        },
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/genres?include[]=total_count',
+      credentials: {}
+    })
+    .then((res) => {
+      expect(res.result.total_count).to.eql(3);
+    });
+  });
+
+  it('does not append the total when the plugin is disabled', () => {
+    server.route({
+      method: 'GET',
+      path: '/books',
+      config: {
+        plugins: {},
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/books?include[]=total_count',
+      credentials: {}
+    })
+    .then((res) => {
+      expect(res.statusCode).to.eql(200);
+      expect(res.result).to.not.have.property('total_count');
+    });
+  });
+
+  it('does not append the total when request validation is not met', () => {
+    server.route({
+      method: 'GET',
+      path: '/books',
+      config: {
+        plugins: {
+          totalCount: { model: Book }
+        },
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/books?include[]=total_count',
+      credentials: null
+    })
+    .then((res) => {
+      expect(res.statusCode).to.eql(200);
+      expect(res.result).to.not.have.property('total_count');
+    });
+  });
+
+  it('errors on incorrect initialization', () => {
+    server.route({
+      method: 'GET',
+      path: '/books',
+      config: {
+        plugins: {
+          totalCount: { model: null }
+        },
+        handler: (request, reply) => {
+          reply({});
+        }
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/books?include[]=total_count',
```

Comment:
what do you think about having a test that doesn't do anything if `?include[]=total_count` isn't included?